lib/actions/upload.js

```diff
@@ -0,0 +1,73 @@
+/*eslint no-invalid-this: 0 no-console: 0*/
+'use strict';
+const co = require('co');
+const Client = require('ssh2-sftp-client');
+const request = require('request');
+const path = require('path');
+const eioUtils = require('elasticio-node').messages;
+const url = require('url');
+
+const sftp = new Client();
+
+function createConnectObject(cfg) {
```

Comment:
I don't think that this function is really needed, its logic could be merged in `init`. If you're going to leave this function then make it sync, please.